doc/source/whatsnew/v0.20.0.txt

```diff
@@ -705,6 +707,138 @@ New Behavior:
    TypeError: Cannot compare 2014-01-01 00:00:00 of
    type <class 'pandas.tslib.Timestamp'> to string column
 
+.. _whatsnew_0200.api_breaking.index_order:
+
+Index order after inner join due to Index intersection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``Index.intersection`` now preserves the order of the calling Index (left)
+instead of the other Index (right) (:issue:`15582`). This affects the inner
+joins (methods ``Index.join``, ``DataFrame.join``, ``DataFrame.merge`` and
+``pd.merge``) and the alignments with inner join (methods ``Series.align`` and
+``DataFrame.align``).
+
+- ``Index.intersection`` and ``Index.join``
+
+  .. ipython:: python
+
+     idx1 = pd.Index([2, 1, 0])
+     idx1
+     idx2 = pd.Index([1, 2, 3])
+     idx2
+
+  Previous Behavior:
+
+  .. code-block:: ipython
+
+     In [4]: idx1.intersection(idx2)
+     Out[4]: Int64Index([1, 2], dtype='int64')
+
+     In [5]: idx1.join(idx2, how='inner')
+     Out[5]: Int64Index([1, 2], dtype='int64')
+
+  New Behavior:
+
+  .. ipython:: python
+
+     idx1.intersection(idx2)
+
+     idx1.join(idx2, how='inner')
+
+- ``Series.align``
+
+  .. ipython:: python
+
+     s1 = pd.Series([20, 10, 0], index=[2, 1, 0])
+     s1
+     s2 = pd.Series([100, 200, 300], index=[1, 2, 3])
+     s2
+
+  Previous Behavior:
+
+  .. code-block:: ipython
+
+     In [4]: (res1, res2) = s1.align(s2, join='inner')
+
+     In [5]: res1
+     Out[5]:
+     1    10
+     2    20
+     dtype: int64
+
+     In [6]: res2
+     Out[6]:
+     1    100
+     2    200
+     dtype: int64
+
+  New Behavior:
+
+  .. ipython:: python
+
+     (res1, res2) = s1.align(s2, join='inner')
+     res1
+     res2
+
+- ``DataFrame.join``, ``DataFrame.merge`` and ``pd.merge``
+
+  .. ipython:: python
+
+     df1 = pd.DataFrame({'a': [20, 10, 0]}, index=[2, 1, 0])
+     df1
+     df2 = pd.DataFrame({'b': [100, 200, 300]}, index=[1, 2, 3])
+     df2
+
+  Previous Behavior:
+
+  .. code-block:: ipython
+
+     In [4]: df1.join(df2, how='inner')
+     Out[4]:
+         a    b
+     1  10  100
+     2  20  200
+
+     In [5]: df1.merge(df2, how='inner', left_index=True, right_index=True)
+     Out[5]:
+         a    b
+     1  10  100
+     2  20  200
+
+     In [6]: pd.merge(df1, df2, how='inner', left_index=True, right_index=True)
+     Out[6]:
+         a    b
+     1  10  100
+     2  20  200
+
+     In [7]: (res1, res2) = df1.align(df2, axis=0, join='inner')
```

Comment:
I don't think it is needed to show both, people know series and dataframe methods work similar. 
I would just list both Series and DataFrame align in the bullet item of Series.align, and then show eg the series example.